### PR TITLE
docs: add keywords to selective loading docs

### PR DESCRIPTION
**Pull request by benmccann**
Should hopefully help with this:

>TIL we can lazy load components in Svelte too. I searched our docs but I don't think we have anything on them. I only first got familiar with the concept when going through the React / Next.js docs where they use a dynamic import with  lazy() or dynamic() .
>
>EDIT: found something briefly mentioned in Kit docs https://kit.svelte.dev/docs/performance#reducing-code-size-selective-loading but it didn't turn up when searching "lazy loading" or "dynamic import"

### CODE REVIEW

**Comment by Rich-Harris** [comment]
It makes no difference. In both cases, searching `dynamic import` brings up this result at the bottom of the list

**Comment by benmccann** [comment]
well it at least helps for searching lazy load I guess

File: documentation/docs/40-best-practices/05-performance.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ To run third party scripts in a web worker (which avoids blocking the main threa
 
 ### Selective loading
 
-Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you need only when some condition is met, use the dynamic `import(...)` form instead.
+Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you want to lazy load or need only when some condition is met, use the dynamic import form instead: `await import(...)`.
```
Comment (Rich-Harris) [inline documentation/docs/40-best-practices/05-performance.md:74]: 'If there is a piece of code you want to lazy load or need only when some condition is met' is pretty awkward grammar — the boundaries of the clauses are unclear, and 'lazy load' should either be 'lazy-load' or (better) 'lazily load'

```suggestion
Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you need only when some condition is met, use the dynamic `import(...)` form to load the component lazily.
```

Comment (benmccann) [reply documentation/docs/40-best-practices/05-performance.md:74]: people aren't going to be searching for "lazily" in the docs, so I switched it to "lazy-load"

Comment (benmccann) [reply documentation/docs/40-best-practices/05-performance.md]: I guess the reason the section title is "selective loading" is that you could be doing it immediately and in that sense it's not lazy. In that sense the suggestion isn't quite accurate, but maybe I'm being overly pedantic about that

**Comment by benmccann** [comment]
Do you need `??=` instead of just `=` in the first example?

For those two be equivalent, this would need to be the `#await` version:
```
<script>
  let showDialog = false;

  function toggleDialog() {
    showDialog = !showDialog;
  }
</script>

{#if showDialog}
  {#await import('$lib/Dialog.svelte') then { default: Dialog }}
    <Dialog></Dialog>
  {/await}
{/if}
```

I think the first version would probably be better performing because this one would probably re-execute `Dialog.svelte` each time. But the advantage of this one is that `toggleDialog` is synchronous, which might be easier to call from a click handler

**Comment by eltigerchino** [comment]
> Do you need ??= instead of just = in the first example?

Maybe it isn’t needed. I wasn’t sure if the reassignment would affect something because of the let syntax reactivity in Svelte 4.

> But the advantage of this one is that `toggleDialog` is synchronous, which might be easier to call from a click handler

Are async event handlers not recommended? Or is it just in this case where we want to display the dialog, which is not some long running task?

**Comment by benmccann** [comment]
> Are async event handlers not recommended?

They're basically fine. There is an eslint rule that warns about them, but they're not really an issue

File: documentation/docs/40-best-practices/05-performance.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ To run third party scripts in a web worker (which avoids blocking the main threa
 
 ### Selective loading
 
-Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you need only when some condition is met, use the dynamic `import(...)` form instead.
+Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you need only when some condition is met, use the dynamic `import(...)` form to lazy-load the component.
```
Comment (benmccann) [inline documentation/docs/40-best-practices/05-performance.md:74]:
```suggestion
Code imported with static `import` declarations will be automatically bundled with the rest of your page. If there is a piece of code you need only when some condition is met, use the dynamic `import(...)` form to selectively lazy-load the component.
```

**Comment by benmccann** [comment]
> Should we include an example of this? I was googling around to learn more and I've seen a few patterns

There's also a long discussion about this in an old PR the FAQ page. When you handle all the different cases it gets quite long and is probably worthy of its own documentation page on the main svelte.dev docs that we can then link to from these various places. Maybe I'll take a stab at that once the docs restructuring is done. For now I'll merge this to clear out the PR queue

**Pull request merged**